### DISCUSSION TITLE
AI 小测生成工具

### DISCUSSION THREAD

**Discussion by QixingQstar** [NonceGeek/indiehacker-handbook#15] [IdeaStack]
我的灵感是从 Huberman Labs 这一期播客来的：https://www.hubermanlab.com/episode/optimal-protocols-for-studying-learning

这篇播客讲了很多有关优化学习效果的理论和方法，我强烈推荐大家有时间也去听一听。播客里面提到的核心的方法是：**在学习完新知识后，立马进行测试，会对学习效果有巨大的提升。**

这让我联想到，人们平时在碎片化时间也会看到读到很多学习资料，譬如 B 站和油管上面的视频、某个人的博客文章、某条长推文、某期播客等等。但是绝大部分人尽管看了很多视频，读了很多文章，里面的知识能记住的却很少。而能够在真实场景检索出来需要的知识并运用的人，就更少了。

基于此，我想设计一个 AI 产品来帮助学习者们可随时基于任意模态的学习资料（集），进行科学的，适合自己学习目标与难度的测验，并得到个性化的积极反馈（text-chat-based or voice-chat-based），从而实现学习效率和兴趣的大幅提升。

以碎片化学习场景为例，这个 AI 产品应该将一篇长文或者一个学习视频分析拆分成不同 checkpoint/quiz point/检查点，每个检查点都生成几道小测题目供学习者使用在这个检查点前学到的知识，去回答这个 checkpoint 的 mini-quiz 里的几个问题。回答成功，就进入到下一部分知识内容的学习；回答得不是很理想，我的 AI 产品也应该像一位老师或者学伴那样，主动启发用户去自主回顾与思考这部分的学习内容，并回答学习者在此期间的疑问，直到辅助学习者完成这个检查点的测验，不像以前似懂非懂地就进入到下一部分内容的学习。

我想，如果用最简单的一句话来总结我设想的产品 MVP 版本，那应该是”把碎片化学习内容 MOOC 化（带助教的那种）“。

**Comment by leeduckgo** [comment]
> 我的灵感是从 Huberman Labs 这一期播客来的：https://www.hubermanlab.com/episode/optimal-protocols-for-studying-learning
> 
> 这篇播客讲了很多有关优化学习效果的理论和方法，我强烈推荐大家有时间也去听一听。播客里面提到的核心的方法是：**在学习完新知识后，立马进行测试，会对学习效果有巨大的提升。**
> 
> 这让我联想到，人们平时在碎片化时间也会看到读到很多学习资料，譬如 B 站和油管上面的视频、某个人的博客文章、某条长推文、某期播客等等。但是绝大部分人尽管看了很多视频，读了很多文章，里面的知识能记住的却很少。而能够在真实场景检索出来需要的知识并运用的人，就更少了。
> 
> 基于此，我想设计一个 AI 产品来帮助学习者们可随时基于任意模态的学习资料（集），进行科学的，适合自己学习目标与难度的测验，并得到个性化的积极反馈（text-chat-based or voice-chat-based），从而实现学习效率和兴趣的大幅提升。
> 
> 以碎片化学习场景为例，这个 AI 产品应该将一篇长文或者一个学习视频分析拆分成不同 checkpoint/quiz point/检查点，每个检查点都生成几道小测题目供学习者使用在这个检查点前学到的知识，去回答这个 checkpoint 的 mini-quiz 里的几个问题。回答成功，就进入到下一部分知识内容的学习；回答得不是很理想，我的 AI 产品也应该像一位老师或者学伴那样，主动启发用户去自主回顾与思考这部分的学习内容，并回答学习者在此期间的疑问，直到辅助学习者完成这个检查点的测验，不像以前似懂非懂地就进入到下一部分内容的学习。
> 
> 我想，如果用最简单的一句话来总结我设想的产品 MVP 版本，那应该是”把碎片化学习内容 MOOC 化（带助教的那种）“。

这个真的很不错👍🏻

**Comment by leeduckgo** [comment]
现在可以思考从哪个领域的小测生成开始切入，可以从你自己感兴趣学习的领域。

**Reply by QixingQstar** [reply]
我比较感兴趣的是个人成长，心理学，认知科学，财经，技术类的话题。

在开发 MVP 的时候，可以用：
1. 我感兴趣的话题
2. 拥有比较多的粉丝群的知识博主

两者有交集的内容，作为开发的测试集或者 Demo 集。这样在后续推广产品的时候，可以有更庞大的关注群体。

我目前能想到一些博主或者 up 主有：
1. Huberman Labs Podcasts
2. Paul Graham Essays
3. “L 先生说”公众号文章
4. “ByteByteGo” weekly public tech article
5. Ali Abdaal
6. 小 Lin 说
7. 一口新饭

其中 1 是播客，能够转换成纯文本且几乎不丢失信息。
2 直接是纯文本，不丢失任何信息。
3, 4 是图文，3 的图少一些，仍以文字为主。4 的图会多不少，且图片会更复杂和更重要。尽管图文都是当前模型能够处理的，只是 MVP 版本从实现速度和成本的角度考虑，可能会只处理文字。
5，6，7 均是视频，转成纯文本会丢失一部分未在文本提及的重要视频图像信息。

----

欢迎大家推荐你们认为对 AI 小测产品开发和推广有价值的学习资料！我非常需要这方面的建议！

**Reply by QixingQstar** [reply]
补充 1：对经典论文（譬如比特币白皮书）和最近的热门论文的 Quiz 生成
补充 2：官方教程的 Quiz 生成，譬如用户以阅读 React 官方文档作为学习 React 的重要教程，每读一篇都来一个测试。
补充 3：企业内部 on-board 文档的 Quiz 生成。这个可能偏 2B 一点，可以先作为 backlog。

**Comment by QixingQstar** [comment]
[AI 小测的最少必要功能点]

给定一段长文字教学资料（可以是博客文章，可以是播客或者视频的 Transcript）:

1. 学习资料预处理
  a. 能够找到该段教学资料的核心概念和关键观点
  b. 能够对该段教学资料进行较为准确的检查点切分

2. 对每个检查点间的内容进行小测生成
  a. 需要一个提示用户这里有个小测检查点的 UI 和用户和小测 AI 进行交互的 UI。目前的想法是像 Readwise Reader 那样，既有一个浏览器插件，使得用户能在原网页上直接进行小测提示和小测交互；也有一个单独的网站，方便用户复习和管理已经做过的小测。但是考虑到做 MVP 的时间限制，我估计优先实现在原网页上直接进行小测交互的浏览器插件开发。

3. 用户作答后，AI 对答案的批改和给用户反馈

4. 用户可以一键导出指定学习资料已经做过的小测
  a. 这个功能的优先级没有前面三点高，但我感觉 MVP 加上它会更完整些，所以这部分 UI 可以做得极简，只要用户知道有导出功能并会用即可。
